packages/apps/esm-login-app/src/location-picker/location-picker.resource.ts

```diff
@@ -0,0 +1,119 @@
+import {
+  setUserProperties,
+  showToast,
+  useSession,
+} from "@openmrs/esm-framework";
+import { useCallback, useEffect, useMemo, useState } from "react";
+import { useTranslation } from "react-i18next";
+import { useValidateLocationUuid } from "../login.resource";
+import { useSearchParams } from "react-router-dom";
+
+export function useDefaultLocation() {
+  const { t } = useTranslation();
+  const [searchParams] = useSearchParams();
+  const [savePreference, setSavePreference] = useState(false);
+  const { user } = useSession();
+  const { userUuid, userProperties } = useMemo(
+    () => ({
+      userUuid: user?.uuid,
+      userProperties: user?.userProperties,
+    }),
+    [user]
+  );
+
+  const defaultLocation = useMemo(() => {
+    return userProperties?.defaultLocation;
+  }, [userProperties?.defaultLocation]);
+
+  const { isLocationValid, defaultLocation: defaultLocationFhir } =
+    useValidateLocationUuid(defaultLocation);
+
+  useEffect(() => {
+    if (defaultLocation) {
+      setSavePreference(true);
+    }
+  }, [setSavePreference, defaultLocation]);
+
+  const updateUserPropsWithPreference = useCallback(
+    async (locationUuid: string, saveUserPreference: boolean) => {
```

Comment:
I think it's bad to call this a "preference" because it's a "default" not a "preference" per se, i.e., if this is checked, next time they login it will go to this location.

```suggestion
    async (locationUuid: string, saveDefaultLocation: boolean) => {
```